### DISCUSSION TITLE
wire:loading issue

### DISCUSSION THREAD

**Discussion by ev3rlost** [livewire/livewire#7369] [1. Help] [answered]
I've been stuck on this one for a while now.

I have two spans in my button - one shows the text "Login", the other a animated spinner. All wired up with wire:loading, showing/hiding accordingly.

The problem occurs when the login method succeeds, the page redirects using `navigate: true`, both spans dissapear, leaving a 'blank button' before navigating to the dashboard.

```
    // login.blade.php

    <form wire:submit="authenticate">
        <button type="submit">
            <span wire:loading.remove>Login</span>
            <span wire:loading><x-spinner /></span>
        </button>
    </form>
```

```
    // Login.php

    public function authenticate()
    {
        $credentials = $this->validate();

        if (Auth::attempt($credentials)) {
            request()->session()->regenerate();
            $this->redirect(route('dashboard'), navigate: true);
        } else {
            $this->addError('denied', 'Wrong credentials!');
        }
    }
```
Normal state
![login-1](https://github.com/livewire/livewire/assets/20288923/0dc8b0e7-e496-4b72-b4e5-249921ec4cea)
When loading
![login-2](https://github.com/livewire/livewire/assets/20288923/a1d334db-326e-4ffc-a994-36b6f5e0334b)
When login succeeds
![login-3](https://github.com/livewire/livewire/assets/20288923/b1aaafbd-364f-4166-997b-1f954296aee3)

Should it not show the normal state again? Or am I missing something ...

**Comment by danie-ramdhani** [comment]
Alternative solution:
Use `wire:loading.class="hidden"` (if you using tailwind css) instead `wire:loading.remove`

**Reply by ev3rlost** [reply]
Thanks! That seemed to work.

**Comment by calebporzio** [comment, marked as answer]
Hey @ev3rlost, turns out what you were experiencing is actually a bug. I fixed it: https://github.com/livewire/livewire/pull/8218

**Reply by ev3rlost** [reply]
Awesome! Thank you Caleb! :) *thumbsup*
